client/src/pages/Book.jsx

```diff
@@ -1,18 +1,30 @@
 import {useState, useEffect} from 'react'
-import { useParams } from 'react-router-dom';
+import { useParams, useLocation } from 'react-router-dom';
 import {MdFavorite, MdFavoriteBorder } from 'react-icons/md';
+import { useAuth0 } from "@auth0/auth0-react";
+
 
 export default function Book() {
 
+    const { isAuthenticated, user } = useAuth0();
     const [book, setBook] = useState([]);
 
     let { id } = useParams();
 
+    //coming from Link
+    const location = useLocation();
+    const { faved, shelf_status } = location.state || {} ;
+ 
+    const [action, setAction] = useState({
+        "isFaved" : faved,
+        "status" : ""
+    })
```

Comment:
I don't quite understand why this is coming from the Link's state. Could this not be handled in `getBookById`?

Reply:
`getBookId` retrieves a book from `demo_api` table that contains only book info. Favorite and status are user actions from the feed table. Since I already had this information on the other page, I accessed them via Link instead of making a new fetch call.